### DISCUSSION TITLE
Set orientation and fill area with hexagons

### DISCUSSION THREAD

**Discussion by thomasreiser** [ManevilleF/hexx#133] [Q&A]
Lets assume I have one area (size: 1000x800):

``+------------------------------+``
``|  (0,0)             (1000,0)  |``
``|                              |``
``|                              |``
``|  (0, 800)        (1000,800)  |``
``+------------------------------+``

What I now want to do is to fill this area completely with a hex grid. So the cell 0,0 should be at position (0,0) and the hex cell with the "biggest" coordinates should be at position (1000,800).

So what I would need would be:
- Ability to setup the hex grid to be aligned top-left (is currently always(?) center)
- Ability to calculate how many hexes I would need to fill the rectangle completely

Is this already possible with the library or do I need to develop something on my own?

**Comment by ManevilleF** [comment]
I would suggest looking add the [`shapes`](https://docs.rs/hexx/latest/hexx/shapes)  module, something like [flat_rectangle](https://docs.rs/hexx/latest/hexx/shapes/fn.flat_rectangle.html).

The functions are very simple, so you can always create a new one from there

**Comment by alice-i-cecile** [comment]
It may be worth enabling Discussions on this repo so we can migrate questions like this into them :)
